terraform/modules/psn/main.tf

```diff
@@ -82,3 +82,18 @@ resource "aws_route53_record" "psn_regioned_service" {
   records = [aws_vpc_endpoint.psn_service.dns_entry[count.index + 1]["dns_name"]]
 }
 
+data "aws_network_interface" "psn_interfaces" {
+  count = length(aws_vpc_endpoint.psn_service.network_interface_ids)
+
+  id = aws_vpc_endpoint.psn_service.network_interface_ids[count.index]
+}
+
+data "template_file" "psn_network_policy_yaml" {
+  template = file("${path.module}/data/psn-network-policy.yaml")
+
+  psn_cidrs = jsonencode(formatlist("%s/32", data.aws_network_interface.psn_interfaces[*].private_ip))
+}
+
+output "psn_network_policy_yaml" {
+  value = data.template_file.psn_network_policy_yaml.rendered
```

Comment:
in terraform 0.12 maybe we can just `value = templatefile("${path.module}/data/psn-network-policy.yaml", {psn_cidrs = jsonencode(formatlist("%s/32", data.aws_network_interface.psn_interfaces[*].private_ip))})` I think, and skip the extra template_file resource?

Reply:
Ew. That's horrible syntax. But the documentation says it's the preferred way now, so I've changed it.